freenas/cli/plugins/openvpn.py

```diff
@@ -0,0 +1,473 @@
+    def __init__(self, name, context):
+        super(OpenVPNManageNamespace, self).__init__(name, context)
+
+        self.entity_subscriber_name = 'service'
+        self.extra_query_params = [('name', '=', 'openvpn')]
+        self.primary_key = self.get_mapping('name')
+
+        self.allow_create = False
+        self.allow_edit = False
+
+        self.add_property(
+            descr='Service name',
+            name='name',
+            get='name',
+            usage=_("""\
+            Name of the service. Read-only value assigned by
+            the operating system."""),
+            set=None,
+            list=True
+        )
+
+        self.add_property(
+            descr='State',
+            name='state',
+            get='state',
+            usage=_("""\
+            Indicates whether the service is RUNNING or STOPPED.
+            Read-only value assigned by the operating system."""),
+            set=None,
+            list=True
+        )
+
+        self.add_property(
+            descr='Process ID',
+            name='pid',
+            get='pid',
+            usage=_("""\
+            Process ID of the RUNNING service. Read-only value assigned
+            by the operating system."""),
+            set=None,
+            list=True
+        )
+
+        self.extra_commands = {
+            'start': OpenVPNManageCommand('start'),
+            'stop': OpenVPNManageCommand('stop'),
+            'reload': OpenVPNManageCommand('reload'),
+            'restart': OpenVPNManageCommand('restart')
+        }
+
+
+
+@description("Start/stop/restart/reload a service")
+class OpenVPNManageCommand(Command):
+    """
+    Usage: start, stop, restart, reload
+    start - starts a service
+    stop - stops a service
+    restart - restarts a service
+    reload - gracefully restarts a service
+    """
+    def __init__(self, action):
+        self.action = action
+
+    @property
+    def description(self):
+        return '{0}s service'.format(self.action.title())
+
+    def run(self, context, args, kwargs, opargs):
+        self.service_id = context.entity_subscribers['service'].query(('name', '=', 'openvpn'),
+                                                                      single=True, select='id')
+        context.submit_task(
+            'service.manage',
+            self.service_id,
+            self.action,
+            )
+
+
+@description("Configure and manage OpenVPN service")
+class OpenVPNamespace(Namespace):
+    """
+    The OpenVPN namespace provides commands for managing,
+    and configuring the OpenVPN service.
+    """
+    def __init__(self, name, context):
+        super(OpenVPNamespace, self).__init__(name)
+        self.context = context
+
+    def namespaces(self):
+        return [
+            OpenVPNManageNamespace('manage', self.context),
+            OpenVPNConfigNamespace('config', self.context)
+        ]
+
+
+def _init(context):
+    context.attach_namespace('/', OpenVPNamespace('openvpn', context))
```

Comment:
I'm not sure if I like `openvpn` being top-level namespace. It belongs to `service` namespace.